src/aap_eda/core/models/activation.py

```diff
@@ -98,152 +90,92 @@ class Meta:
         related_name="+",
     )
     awx_token = models.ForeignKey(
-        AwxToken,
+        "AwxToken",
         on_delete=models.SET_NULL,
         null=True,
         default=None,
     )
     sources = models.ManyToManyField(
-        Source,
+        "Source",
         default=None,
     )
 
-    def save(self, *args, **kwargs):
-        # when creating
-        if self._state.adding:
-            if self.status_message is None:
-                self._set_status_message()
-        else:
-            if not bool(kwargs) or "update_fields" not in kwargs:
-                raise UpdateFieldsRequiredError(
-                    "update_fields is required to use when saving "
-                    "due to race conditions"
-                )
-            else:
-                if "status" in kwargs["update_fields"]:
-                    self._is_valid_status()
-
-            if (
-                "status_message" in kwargs["update_fields"]
-                and "status" not in kwargs["update_fields"]
-            ):
-                raise StatusRequiredError(
-                    "status_message cannot be set by itself, "
-                    "it requires status and status_message together"
-                )
-            # when updating without status_message
-            elif (
-                "status" in kwargs["update_fields"]
-                and "status_message" not in kwargs["update_fields"]
-            ):
-                self._set_status_message()
-                kwargs["update_fields"].append("status_message")
-
-        super().save(*args, **kwargs)
-
-    def _set_status_message(self):
-        self.status_message = self._get_default_status_message()
-
-        if self.status == ActivationStatus.PENDING and not self.is_enabled:
-            self.status_message = "Activation is marked as disabled"
-
-    def _get_default_status_message(self):
-        try:
-            return ACTIVATION_STATUS_MESSAGE_MAP[self.status]
-        except KeyError:
-            raise UnknownStatusError(f"Status [{self.status}] is invalid")
-
-    def _is_valid_status(self):
```

Comment:
@Alex-Izquierdo Where did this move to?

Reply:
In the mixin: https://github.com/ansible/eda-server/pull/617/files#diff-23367b3fab1d54eb173437e73363840615b77ca82575d88ede9a996520d4cc67R111